doc/debuginfo.txt

```diff
@@ -0,0 +1,187 @@
+    [19:53:41.648423786] (+0.000029376) colossus my_provider:my_first_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB5A6, debug_info = { func = "bar", source_loc = "/home/efficios/example/libhello.c:13" }, vpid = 25577 }, { my_string_field = "recoltes et semailles", my_integer_field = 57 }
+    [19:53:41.648435726] (+0.000011940) colossus my_provider:my_other_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB66B, debug_info = { func = "baz", source_loc = "/home/efficios/example/libhello.c:20" }, vpid = 25577 }, { some_field = 1729 }
+    [...]
+
+The interesting part is the debug_info section of the context:
+
+    debug_info = { func = "foo", source_loc = "/home/efficios/example/libhello.c:7" }
+
+This is the expected output for events generated by an executable for
+which DWARF information is available. It shows the name of the
+function containing the tracepoint instance which generated the event
+("foo"), and its source location ("libhello.c", line 7).
+
+The second event in the sample output is of the same type
+("my_first_tracepoint"), but it was generated by a different
+tracepoint instance, hence the different source location (line 13) and
+function ("bar").
+
+The third event, of a different type, also shows debug information.
+
+If DWARF info is absent, but ELF symbols are not stripped, the output
+will instead look like this:
+
+   [...]
+   [19:53:41.648394410] (+0.000022616) colossus my_provider:my_first_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB4D4, debug_info = { func = "foo+0xa9" }, vpid = 25577 }, { my_string_field = "hello, tracer", my_integer_field = 42 }
+   [19:53:41.648423786] (+0.000029376) colossus my_provider:my_first_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB5A6, debug_info = { func = "bar+0xa9" }, vpid = 25577 }, { my_string_field = "recoltes et semailles", my_integer_field = 57 }
+   [19:53:41.648435726] (+0.000011940) colossus my_provider:my_other_tracepoint: { cpu_id = 2 }, { ip = 0x7F75AEDFB66B, debug_info = { func = "baz+0x9c" }, vpid = 25577 }, { some_field = 1729 }
+   [...]
+
+The debug information now provides only the function name as given by
+ELF symbols, plus an offset in bytes from the symbol to the actual
+location of the tracepoint instance.
+
+Debug Info and Dynamic Loading
+------------------------------
+
+Babeltrace can resolve addresses of events originating from
+dynamically loaded libraries, provided some extra information is
+collected at tracing time.
+
+This can be achieved by preloading LTTng UST's libdl helper when
+launching the program to be traced, like so:
+
+    $ LD_PRELOAD="liblttng-ust-dl.so" <path/to/executable>
+
+The tracing and analysis can now be performed as described in prior
+sections, and events from tracepoints in dlopened libraries will be
+resolved automatically by babeltrace.
+
+Separate Debug Info
+-------------------
+
+It is possible to have DWARF debug information separate from an
```

Comment:
I did change "have" to "store", but I feel that "separate" is the proper usage here as it serves the role of adjective as in "separate from an executable", rather than that of adverb, as if the emphasis were on "to store separately".